frontend/src/components/hamburger/hamburger-sidemenu.tsx

```diff
@@ -0,0 +1,79 @@
+import { Link } from "@tanstack/react-router";
+import HamburgerOverlay from "./hamburger-overlay";
+
+
+interface HamburgerMenuProps {
+    isMenuOpen: boolean;
+    setIsMenuOpen: React.Dispatch<React.SetStateAction<boolean>>;
+}
+
+const HamburgerMenu = ({ isMenuOpen, setIsMenuOpen }: HamburgerMenuProps) => {
+    const tabs = ["Home", "About", "Games", "Team", "Contact"];
```

Comment:
So the cool thing about our routing package (Tanstack Router) is that it will implement typesafety on our routes. The way you've set it up here is not typesafe. Actually, `About` isn't even a route! Going to this route will lead to a not found page.

We can setup typesafety like so. First we need to import the right interfaces and declare a type.

```tsx
import type { FileRouteTypes } from "../routeTree.gen";

type Route = FileRouteTypes["to"];
```

Then we can do the following
```suggestion
    const tabs: Record<string, Route> = {
      Home: '/',
      Games: '/games',
      Team: '/team',
      Contact: '/contact',
    }
```